File: ans.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
		int tc,n,i,j;
	int a[1000001]={0};
	int least_prime[1000001];
	for(i=0;i<=1000000;i++)
	{
		least_prime[i]=i;
	}
	for(i=2;i<=1000000;i++)
	{
		if(a[i]==0)
		{
			for(j=2*i;j<=1000000;j=j+i)
			{
				a[j]=1;
				if(i<least_prime[j])
				{
					least_prime[j]=i;
				}
			}
		}
	}

	cin>>tc;
	while(tc--)
	{
		cin>>n;
		cout<<n-least_prime[n]<<" "<<least_prime[n]<<endl;
		
	}
	
	
}

File: ans.py
a=[]
for i in range(1000000):
 a.append(0)

least_prime=[]
for i in range(1000000):
 least_prime.append(i)
 
for i in range(2,1000000):
 if a[i]==0:
     for j in range(2*i,1000000,i):
         a[j]=1
         if i<least_prime[j]:
          least_prime[j]=i
          
tc=int(raw_input())
for i in range(tc):
 n=int(raw_input())
 print "%d %d" % (n-least_prime[n],least_prime[n])